wordpress-mobile/WordPress-Android
# Prevent contributors from uploading media files

Fixes #10666

I recommend using another diff tool than GitHub - eg. the changes in Android Studio diff are more obvious.

The goal of this PR is to prevent users who don't have permission, to upload media files from the Editor (eg. Contributors). Ideal solution wouldn't even show the items on the UI (eg. we wouldn't show "Capture image" as we know, it'll require media upload). I decided to implement this temporary solution and we can consider reverting it when we add support for filtering media options before Gutenberg shows them on the UI. We could use similar technique to [this](https://github.com/wordpress-mobile/WordPress-Android/blob/c9e3eb11b886f782236549e97ed15a3aacf56fc7/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java#L288) for all options we show on the UI.  Wdyt?



To test:
1. Switch to a site on which you are in "Contributor" role
2. Create a new draft in Gutenberg
3. Try to upload media - Image block, Video block, Media&Text block
=> the only option which should work is WPMedia library as the media files are already present on the server


PR submission checklist:
